engine/runtime/src/memories/frag.rs

```diff
@@ -0,0 +1,427 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module provides functionality to allocate memory with higher randomness on returned addresses.
+//!
+//! The most simple approach is to allocate memory multiple times, return the final allocation
+//! as the desired memory.
+//!
+//! Allocators differ between operating systems. On *nix and BSD-based systems `malloc(int)` is being called
+//! but does not initialize the allocated spaced. Microsoft Windows uses
+//! [`VirtualAlloc`](https://docs.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-virtualalloc)
+//! allocates and initializes a region of memory pages with zeroes.
+//!
+//! [`FragStrategy`] implements at least two possible allocation strategies:
+//!
+//! - Direct: The algorithm tries to allocate a huge amount of memory space while keeping a certain address distance
+//! - Memory Mapped: anonymous memory is being mapping, the memory address will be randomly selected.
+
+use crate::MemoryError;
+use log::*;
+use std::{fmt::Debug, ptr::NonNull};
+
+/// Fragmenting strategy to allocate memory at random addresses.
+#[derive(Debug, Clone, Copy)]
+#[non_exhaustive]
+pub enum FragStrategy {
+    /// Anonymously maps a region of memory
+    Map,
+
+    /// System's allocator will be called a few times
```

Comment:
Which one are you referring to? Otherwise I would say the `FragStrategry` is exactly that: `allocate memory at random addresses`. Maybe an addendum is missing. Something like "[...] that are reasonably far apart" Even though "reasonably" cannot be expressed numerically :shrug:

Reply:
"System's allocator will be called a few times". 
And now that I've read the pr the `Direct` strategy doesn't do that currently. Now it allocates a big chunk and hide the relevant memory within

Reply:
what do you mean? The "algorithm" loops over the `malloc` and calls it again, if it fails.